src/Modal/index.jsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+import styles from './Modal.scss';
+import Button, { buttonPropTypes } from '../Button';
+
+
+class Modal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.addEvents = this.addEvents.bind(this);
+    this.removeEvents = this.removeEvents.bind(this);
+    this.handleDocumentClick = this.handleDocumentClick.bind(this);
+    this.close = this.close.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.buttons = [];
+
+    // is there a way to avoid depending on the props for the open state?
+    this.state = {
+      open: props.open,
+      focusIndex: 0,
+    };
+  }
+
+  componentDidMount() {
+    this.buttons[0].focus();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.open !== this.props.open) {
+      this.setState({ open: nextProps.open });
+    }
+  }
+
+  componentWillUpdate(_, nextState) {
+    if (nextState.open) {
+      this.addEvents();
+    } else {
+      this.removeEvents();
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.state.open) {
+      this.buttons[this.state.focusIndex].focus();
+    }
+  }
+
+  close() {
+    this.setState({ open: false });
+    this.props.onClose();
+  }
+  // what does true mean here?
+  addEvents() {
+    document.addEventListener('click', this.handleDocumentClick, true);
+  }
+
+  removeEvents() {
+    document.removeEventListener('click', this.handleDocumentClick, true);
+  }
+
+  handleDocumentClick(e) {
```

Comment:
Right now, this just ensures that the focusIndex is updated when someone clicks on a button and the focus moves to that button.